### PR TITLE
Addresses #38, combineReducers can handle custom data structures.

### PR DESCRIPTION
**Pull request by leonaves**
I returned to a reduce function on the object.keys, as I don't see it being any less efficient, and personally I much prefer it to a for loop, but I kept some of the changes from #36 which did increase efficiency. Let me know what you think.

### CODE REVIEW

File: modules/combineReducers.js [hunk 1]
```diff
@@ -12,32 +12,36 @@ function optimizeBatch(effects) {
   }
 }
 
-export function combineReducers(reducerMap) {
-  return function finalReducer(state = {}, action) {
-    let hasChanged = false;
-    let effects = [];
-    let model = {};
+export function combineReducers(reducerMap, rootState = {}, accessor, mutator) {
+    if (typeof accessor !== 'function') {
```
Comment (lukewestby) [inline modules/combineReducers.js]: can you lift the default `accessor` and `mutator` functions out of the `combineReducers` body and use default args to assign them? could be a little bit of a perf boost to not create new default functions on every call

**Comment by leonaves** [comment]
How's that, I'm not too familiar with code style for long argument lists in javascript so I just followed [PSR-2](http://www.php-fig.org/psr/psr-2/#4-4-method-arguments) 😜 I won't be offended if you reformat it.

**Comment by leonaves** [comment]
How's that? I'm not too familiar with code style for long argument lists in javascript so I just followed [PSR-2](http://www.php-fig.org/psr/psr-2/#4-4-method-arguments) 😜 I won't be offended if you reformat it.

**Comment by lukewestby** [comment]
Code style looks good! Can you install `immutable` as a devDependency and write a test case for the new functionality to show that (a) the default implementation works the same and (b) your new use case with Immutable works like you expect? You can put those tests in a separate file `combineReducers.test.js`

As a side note, does `Object.keys(obj)` work on an `Immutable.Map()`?

**Comment by leonaves** [comment]
Okay, getting on that now, and now `Object.keys(obj)` does not work on `Immutable.Map()`, no, it will give the keys of the root object, e.g.:

```javascript
[ 'size',
  '_root',
  '__ownerID',
  '__hash',
  '__altered' ]
```

But that's okay, `reducerMap` should be an object anyway, it's just the state itself that may be another data structure.

**Comment by leonaves** [comment]
You could change the `Object.keys(reducerMap).reduce()` to a [`for...of` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of), which would support a reducerMap of any data structure that implements the Iterable interface (including `Immutable.Map()`) but I'm not sure how necessary that is.

**Comment by lukewestby** [comment]
Aha! I'm following now, I think we'll stick with the plain object reducer map. Thanks for clearing that up. This looks good to me! I'm planning to cut a release tomorrow, I'll make sure this gets merged in for that.

**Comment by lukewestby** [comment]
Awesome! @leonaves this is an exemplary OSS contribution, thanks so much.

**Pull request merged**